src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/SvType.java

```diff
@@ -1,41 +1,166 @@
 package org.broadinstitute.hellbender.tools.spark.sv.discovery;
 
 import htsjdk.variant.variantcontext.Allele;
+import htsjdk.variant.variantcontext.VariantContextBuilder;
+import htsjdk.variant.vcf.VCFConstants;
+import org.apache.commons.lang3.EnumUtils;
+import org.broadinstitute.hellbender.engine.datasources.ReferenceMultiSource;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.spark.sv.discovery.inference.NovelAdjacencyAndAltHaplotype;
+import org.broadinstitute.hellbender.tools.spark.sv.utils.GATKSVVCFConstants;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
 
-import java.util.Collections;
-import java.util.Map;
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.broadinstitute.hellbender.tools.spark.sv.utils.GATKSVVCFConstants.INTERVAL_VARIANT_ID_FIELD_SEPARATOR;
 
 
 /**
- * Various types of structural variations
+ * Various types of structural variations.
+ * Holding minimum information on VariantContext
+ * it can hold:
+ * CHR, POS, ID, REF, ALT, SVTYPE, END (optional), SVLEN(optional)
  */
 public abstract class SvType {
+    public static final int NO_APPLICABLE_END = -1;
```

Comment:
As mentioned above, what if we replaced these with methods `hasEnd` and `hasLength`?

Reply:
done